corehq/apps/sms/api.py

```diff
@@ -719,6 +731,8 @@ def _process_incoming(msg):
                 send_sms_with_backend(msg.domain, msg.phone_number, text, msg.backend_id, metadata=metadata)
             else:
                 send_sms(msg.domain, None, msg.phone_number, text, metadata=metadata)
+            if _domain_accepts_inbound(msg) and _load_handlers(v, is_two_way, has_domain_two_way_scope):
```

Comment:
Instead of adding these two lines to the first two branches, can you run them after the if/else? Looks like the `process_pre_registration` and `process_sms_registration` would still need to be run afterwards if it isn't an opt message, but I think it would still be less code overall.